server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -451,6 +447,195 @@ public void testFilterBySLMPolicy() throws Exception {
         assertThat(getAllSnapshotsForPolicies(GetSnapshotsRequest.NO_POLICY_PATTERN, "*"), is(allSnapshots));
     }
 
+    public void testSortAfterStartTime() {
+        final String repoName = "test-repo";
+        createRepository(repoName, "fs");
+        final SnapshotInfo snapshot1 = createFullSnapshot(repoName, "snapshot-1");
+        final SnapshotInfo snapshot2 = createFullSnapshot(repoName, "snapshot-2");
+        final SnapshotInfo snapshot3 = createFullSnapshot(repoName, "snapshot-3");
+
+        final List<SnapshotInfo> allSnapshotInfo = clusterAdmin().prepareGetSnapshots(matchAllPattern())
+            .setSnapshots(matchAllPattern())
+            .setSort(GetSnapshotsRequest.SortBy.START_TIME)
+            .get()
+            .getSnapshots();
+        assertThat(allSnapshotInfo, is(List.of(snapshot1, snapshot2, snapshot3)));
+
+        final long startTime1 = snapshot1.startTime();
+        final long startTime2 = snapshot2.startTime();
+        final long startTime3 = snapshot3.startTime();
+
+        assertThat(allAfterStartTimeAscending(startTime1 - 1), is(allSnapshotInfo));
+        assertThat(allAfterStartTimeAscending(startTime1), is(allSnapshotInfo));
+        assertThat(allAfterStartTimeAscending(startTime2), is(List.of(snapshot2, snapshot3)));
+        assertThat(allAfterStartTimeAscending(startTime3), is(List.of(snapshot3)));
+        assertThat(allAfterStartTimeAscending(startTime3 + 1), empty());
+
+        final List<SnapshotInfo> allSnapshotInfoDesc = clusterAdmin().prepareGetSnapshots(matchAllPattern())
+            .setSnapshots(matchAllPattern())
+            .setSort(GetSnapshotsRequest.SortBy.START_TIME)
+            .setOrder(SortOrder.DESC)
+            .get()
+            .getSnapshots();
+        assertThat(allSnapshotInfoDesc, is(List.of(snapshot3, snapshot2, snapshot1)));
+
+        assertThat(allBeforeStartTimeDescending(startTime3 + 1), is(allSnapshotInfoDesc));
+        assertThat(allBeforeStartTimeDescending(startTime3), is(allSnapshotInfoDesc));
+        assertThat(allBeforeStartTimeDescending(startTime2), is(List.of(snapshot2, snapshot1)));
+        assertThat(allBeforeStartTimeDescending(startTime1), is(List.of(snapshot1)));
+        assertThat(allBeforeStartTimeDescending(startTime1 - 1), empty());
+    }
```

Comment:
Can we add a combination test too, validating that `after_value` and for instance snapshot name filtering works together? Just one of these is enough, no need to do several combinations.

Reply:
Added a test for that and also for offset + `after_value` which turned out to be trivial.